### PR TITLE
Fix bug 1588929 - Localize 'All Projects' & 'All resources' in the main navigation bar

### PR DESCRIPTION
**Pull request by katherine95**
#### What does this PR do?
This pull request fixes the issue [1588929](https://bugzilla.mozilla.org/show_bug.cgi?id=1588929). It has "All Resources" and "All Projects"  unaccented (unlocalized) in the main navigation bar:

#### Type of change:
- [x] Bug fix (non-breaking change which fixed an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update.

#### How Has This Been Tested?
- Clone this repository 
- Git checkout to `katherine95:1588929-localize-resources-projects` branch.
- Follow the setup guidelines to set up pontoon locally
- Run `make build` then `make run`
- Login to Pontoon
- Navigate to Resource menu (access it by clicking on "playground" in `http://localhost:8000/projects/tutorial/playground/`)
- "All Resources" and "All Projects" should both be localized
`localhost:8000/projects/tutorial/all-resources/?pseudolocalization=accented`
`localhost:8000/projects/all-projects/all-resources/?pseudolocalization=accented`

#### Checklist:
 [x] Both "All Resources" and "All Projects"  using pseudo-localization should be localized.

#### Screenshots (if appropriate):
Screenshot from 2019-10-24 19-32-19

Screenshot from 2019-10-24 19-49-36

### CODE REVIEW

**Comment by mathjazz** [comment]
That is the opposite of what needs to be done. :)

We need to "make All Projects/Resources in the main navigation localizable" and not "make All Projects/Resources in the resource menu non-localizable".

Please revert this commit.

**Comment by katherine95** [comment]
> That is the opposite of what needs to be done. :)
> 
> We need to "make All Projects/Resources in the main navigation localizable" and not "make All Projects/Resources in the resource menu non-localizable".
> 
> Please revert this commit.

I will revert and make the required changes. I misunderstood you initially.

**Comment by katherine95** [comment]
> > That is the opposite of what needs to be done. :)
> > We need to "make All Projects/Resources in the main navigation localizable" and not "make All Projects/Resources in the resource menu non-localizable".
> > Please revert this commit.
> 
> I will revert and make the required changes. I misunderstood you initially.

Also allow me to fix the failing test

File: frontend/src/core/project/components/ProjectMenu.js [hunk 1]
```diff
@@ -139,7 +139,7 @@ export class ProjectMenuBase extends React.Component<Props, State> {
                 className="selector unselectable"
                 onClick={ this.toggleVisibility }
             >
-                <span>{ 'All Projects' }</span>
+                <span>{ <Localized id= 'resource-ResourceMenu--all-projects'>All Projects</Localized> }</span>
```
Comment (mathjazz) [inline frontend/src/core/project/components/ProjectMenu.js]: Please check out how to use the `Localized` tag properly:
https://github.com/mozilla/pontoon/tree/master/frontend#localization

```html
<Localized id="ID">
    <span>Bla</span>
</Localized>
```

Also, mind the space after `=`.

File: frontend/src/core/resource/components/ResourceMenu.js [hunk 2]
```diff
@@ -117,7 +117,7 @@ export class ResourceMenuBase extends React.Component<Props, State> {
 
         let resourceName = parameters.resource.split('/').slice(-1)[0];
         if (parameters.resource === 'all-resources') {
-            resourceName = 'All Resources';
+            resourceName = <Localized id= 'resource-ResourceMenu--all-resources'>All Resources</Localized>;
```
Comment (mathjazz) [inline frontend/src/core/resource/components/ResourceMenu.js]: Nit: space.

File: frontend/src/core/project/components/ProjectMenu.test.js [hunk 3]
```diff
@@ -58,7 +60,7 @@ describe('<ProjectMenu>', () => {
         const wrapper = createShallowProjectMenu({ project: ALL_PROJECTS });
 
         expect(wrapper.find('.project-menu .selector')).toHaveLength(1);
-        expect(wrapper.find('.project-menu .selector span:first-child').text()).toEqual('All Projects');
+        expect(findLocalizedById(wrapper, 'resource-ResourceMenu--all-projects')).toHaveLength(1);
```
Comment (mathjazz) [inline frontend/src/core/project/components/ProjectMenu.test.js]: Wouldn't that work as well (and be more elegant)?
```js
expect(wrapper.find('#resource-ResourceMenu--all-projects')).toHaveLength(1);
```

**Comment by katherine95** [comment]
> Nice work! Left a few comments.
> 
> Please also update the en-US .ftl file with newly added localizable strings as per the docs:
> https://github.com/mozilla/pontoon/tree/master/frontend#localization

The en-US file already has the localizable strings

File: frontend/src/core/resource/components/ResourceMenu.js (hunk 2 above)
Comment (katherine95) [reply frontend/src/core/resource/components/ResourceMenu.js]: Should I also fix the use of Localized tag here? @mathjazz

File: frontend/src/core/project/components/ProjectMenu.test.js (hunk 3 above)
Comment (katherine95) [reply frontend/src/core/project/components/ProjectMenu.test.js]: sure it will work.

File: frontend/src/core/project/components/ProjectMenu.js [hunk 4]
```diff
@@ -139,7 +139,13 @@ export class ProjectMenuBase extends React.Component<Props, State> {
...
                 onClick={ this.toggleVisibility }
             >
-                <span>{ <Localized id= 'resource-ResourceMenu--all-projects'>All Projects</Localized> }</span>
+                <span>
+                    {
+                        <Localized id='resource-ResourceMenu--all-projects'>
+                            <span>All Projects</span>
+                        </Localized>
+                    }
+                </span>
```
Comment (mathjazz) [inline frontend/src/core/project/components/ProjectMenu.js]: You don't need the outer `<span>` and `{}`.

File: frontend/src/core/resource/components/ResourceMenu.js (hunk 2 above)
Comment (mathjazz) [reply frontend/src/core/resource/components/ResourceMenu.js]: What do you mean by fix?

**Comment by mathjazz** [comment]
> The en-US file already has the localizable strings. I was just re-using what was already added.

Yes, but we should create new entries with new string identifiers, following the documented naming scheme: https://github.com/mozilla/pontoon/tree/master/frontend#localization

File: frontend/src/core/project/components/ProjectMenu.js [hunk 5]
```diff
@@ -139,7 +139,13 @@ export class ProjectMenuBase extends React.Component<Props, State> {
                 className="selector unselectable"
                 onClick={ this.toggleVisibility }
             >
-                <span>{ 'All Projects' }</span>
+                <span>
+                    {
+                        <Localized id='resource-ResourceMenu--all-projects'>
```
Comment (katherine95) [inline frontend/src/core/project/components/ProjectMenu.js]: For some reason, having the string identifier/id `project-ProjectMenu--all-projects ` and adding it to main translation file doesn't localize 'All Projects' in the main navigation. 
Why could it be behaving this way?

File: frontend/src/core/project/components/ProjectMenu.test.js (hunk 3 above)
Comment (mathjazz) [reply frontend/src/core/project/components/ProjectMenu.test.js]: Nice!

**Comment by katherine95** [comment]
> > The en-US file already has the localizable strings. I was just re-using what was already added.
> 
> Yes, but we should create new entries with new string identifiers, following the documented naming scheme: https://github.com/mozilla/pontoon/tree/master/frontend#localization

I am still trying to figure out why the new string identifier `project-ProjectMenu--all-projects` and adding it to main translation file doesn't localize 'All Projects' in the main navigation.

File: frontend/src/core/project/components/ProjectMenu.js (hunk 5 above)
Comment (katherine95) [reply frontend/src/core/project/components/ProjectMenu.js]: make build is taking some time. I will let you know the outcome.

Comment (katherine95) [reply frontend/src/core/project/components/ProjectMenu.js]: I have tried that and the result is still the same, 'All Projects' doesn't get localized

Comment (katherine95) [reply frontend/src/core/project/components/ProjectMenu.js]: It is now fixed.

**Pull request merged**